content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,240 @@
+non-eBPF CNIs with this approach.
+
+### Addressing the challenges
+
+A new solution was necessary - doing redirection of any sort in the node's network namespace would create unavoidable conflicts,
+unless we compromised our compatibility requirements.
+
+In sidecar mode, it is trivial to configure traffic redirection between the sidecar and application pod, as both operate within
+the pod's network namespace. This led to a light-bulb moment: why not mimic sidecars, and configure the redirection in
+the application pod's network namespace?
+
+While this sounds like a "crazy simple" thought, is this even possible, given ztunnel runs
+in the Istio system namespace? After some research, we discovered a Linux process running in one network namespace
+could create and own listening sockets within another network namespace, which is a basic Linux socket capability.
+However, to make this work, and cover all pod lifecycle scenarios we had to make architectural changes to the ztunnel
+as well as the `istio-cni` agent.
+
+After prototyping and sufficiently validating that this innovative approach does work for all the Kubernetes platforms we have
+access to, we built confidence in the work and decided to contribute this new traffic redirection
+model. An *in-Pod* traffic redirection mechanism between workload pods and the ztunnel node proxy component that has
+been built from the ground up to be highly compatible with all major cloud providers and CNIs.
+
+The key innovation is to deliver the pod’s network namespace to the ztunnel so that ztunnel can start its redirection
+sockets inside the pod’s network namespace, while still running outside the pod. With this approach, the traffic redirection
+between ztunnel and application pods happens in a way that’s very similar to sidecars and application pods today and is
+strictly invisible to any Kubernetes CNI. Network policy from any Kubernetes CNIs can continue to be enforced,
+regardless of whether the CNI uses eBPF or iptables, there is no conflict between the in-Pod traffic redirection and the CNI.
+
+## Technical deep dive of in-Pod traffic redirection
+
+First, let’s go over the basics of how a packet travels between pods in Kubernetes.
+
+### Linux, Kubernetes, and CNI  - what’s a network namespace, and why does it matter?
+
+In Linux, a *container* is one or more Linux processes running within isolated Linux namespaces. A Linux namespace
+is simply a kernel flag that controls what processes running within that namespace are able to see. For instance, if you
+create a new Linux network namespace via the `ip netns add my-linux-netns` command and run a process inside it, that process can only see the networking rules created
+within that network namespace. It can not see any network rules created outside of it - even though everything running on that machine is still
+sharing one Linux networking stack.
```

Comment:
```suggestion
within that network namespace. It can not see any network rules created outside of it - even though everything running on that machine is still sharing one Linux networking stack.
```

weird LB (not sure if these matter in rendering)